app/src/main/java/com/github/onlynotesswent/model/folder/FolderViewModel.kt

```diff
@@ -523,11 +538,144 @@ class FolderViewModel(private val repository: FolderRepository) : ViewModel() {
+    // is not yet fetched). Otherwise, use the saved folders list already in the viewModel.
+    if (_userSavedFolders.value.isEmpty()) {
+      getCurrentUserSavedFolders(
+          userViewModel,
+          onSuccess = { savedFolders ->
+            setCurrentUserSavedFolders(
+                userViewModel, savedFolders.filter { it.id != folderId }, onSuccess, onFailure)
+          },
+          onFailure = onFailure)
+    } else {
+      setCurrentUserSavedFolders(
+          userViewModel, _userSavedFolders.value.filter { it.id != folderId }, onSuccess, onFailure)
+    }
+  }
+
+  /**
+   * Retrieves the list of saved folders of the current user.
+   *
+   * @param userViewModel The userViewModel to use for retrieving the saved folders.
+   * @param onSuccess The function to call when the retrieval is successful.
+   * @param onFailure The function to call when the retrieval fails.
+   * @param useCache Whether to update data from cache.
+   */
+  fun getCurrentUserSavedFolders(
+      userViewModel: UserViewModel,
+      onSuccess: (List<Folder>) -> Unit = {},
+      onFailure: (Exception) -> Unit = {},
+      useCache: Boolean = true
+  ) {
+    // Get the list of saved document IDs of type FOLDER for the current user from the
+    // userViewModel,
+    // and retrieve them
+    userViewModel.getSavedDocumentIdsOfType(
+        documentType = UserRepositoryFirestore.SavedDocumentType.FOLDER,
+        onSuccess = { documentIds ->
+          viewModelScope.launch {
+            repository.getSavedFoldersByIds(
+                savedFoldersIds = documentIds,
+                friends = userViewModel.currentUser.value!!.friends,
+                onSuccess = { savedFolders, nonSaveableFoldersIds ->
+                  Log.d("FolderViewModel", "Saved folders: $savedFolders")
+                  _userSavedFolders.value = savedFolders
+
+                  // Delete the no longer saveable (deleted or privated) folders from
+                  // the user's saved folders list
+                  for (folderId in nonSaveableFoldersIds) {
+                    userViewModel.deleteSavedDocumentIdOfType(
+                        documentType = UserRepositoryFirestore.SavedDocumentType.FOLDER,
+                        documentId = folderId)
+                  }
+
+                  onSuccess(savedFolders)
+                },
+                onFailure = onFailure,
+                useCache = useCache)
+          }
+        },
+        onFailure = onFailure)
+  }
+
+  /**
+   * Sets the current user's saved folders list with the given list of folders.
+   *
+   * @param userViewModel The userViewModel to use for updating the user's saved folders list in
+   *   cloud.
+   * @param folders The new list of saved folders for the user.
+   * @param onSuccess The function to call when the addition is successful.
+   * @param onFailure The function to call when the addition fails.
+   */
+  private fun setCurrentUserSavedFolders(
```

Comment:
Because it isn't used or needed, and shouldn't really be used as it completely overrides your saved folders list, which if someone doesn't fully understand what it does could introduce hard to find bugs. Safer IMO to leave it private, and eventually if we do need it we can simply remove the private